### PR TITLE
Updates based on ID changes

### LINKED ISSUE #116: Enable deleting relationships
```[tasklist]
### Tasks
- [ ] https://github.com/nsidc/usaon-vta-survey/issues/121
```

### PR DESCRIPTION
**Pull request by rmarow**

### CODE REVIEW

**Comment by rmarow** [comment]
@MattF-NSIDC I would like to merge this into main since it's actually almost ALL (besides the final commit) work on things related to the update DB setup from https://github.com/nsidc/usaon-vta-survey/pull/122 this PR

File: usaon_vta_survey/models/tables.py [hunk 1]
```diff
@@ -334,10 +342,11 @@ class ResponseApplication(BaseModel, IORelationshipMixin, ResponseObjectFieldMix
     performance_criteria = Column(
         String(256),
     )
-    # limit 0-100
     performance_rating = Column(
         Integer,
-        # CheckConstraint("0<=performance_rating<=100"),
+        CheckConstraint(
+            'performance_rating>0 and performance_rating<101', name='0-100'
```
Comment (MattF-NSIDC) [inline usaon_vta_survey/models/tables.py:348]: I think we should do `>= 0` and `<=100` here for explicitness.

File: usaon_vta_survey/models/tables.py [hunk 2]
```diff
@@ -418,15 +433,28 @@ class ResponseObservingSystemDataProduct(BaseModel):
...
+        SmallInteger,
+        CheckConstraint(
+            'criticality_rating>=0 and criticality_rating<=100', name='c0-100'
+        ),
+        nullable=False,
+    )
+    performance_rating = Column(
+        SmallInteger,
+        CheckConstraint(
+            'performance_rating>=0 and performance_rating<=100', name='0-100'
```
Comment (MattF-NSIDC) [inline usaon_vta_survey/models/tables.py:454]: Nitpick: for consistency with `c0-100`, I think we should use `p0-100` here.

File: usaon_vta_survey/routes/response/relationships/application_societal_benefit_area.py [hunk 3]
```diff
@@ -97,9 +97,15 @@ def _response_application_societal_benefit_area(
...
-        response_application_societal_benefit_area = db.session.get(
-            ResponseApplicationSocietalBenefitArea,
-            (societal_benefit_area_id, application_id),
+        response_application_societal_benefit_area = (
+            db.session.query(ResponseApplicationSocietalBenefitArea)
+            .filter(
+                ResponseApplicationSocietalBenefitArea.response_societal_benefit_area_id
+                == societal_benefit_area_id
+                and ResponseApplicationSocietalBenefitArea.response_application_id
+                == application_id
```
Comment (MattF-NSIDC) [inline usaon_vta_survey/routes/response/relationships/application_societal_benefit_area.py:106]: Ah, an unfortunate down-side of demoting these columns from PK status. :shrug: :)

**Pull request merged**